cli/tsc/dts/lib.esnext.collection.d.ts

```diff
@@ -27,3 +27,34 @@ interface MapConstructor {
         keySelector: (item: T, index: number) => K,
     ): Map<K, T[]>;
 }
+
+interface Set<T> {
+    /**
+     * @returns a new Set containing all the elements in this Set and also all the elements in the argument.
+     */
+    union<U>(other: Set<U>): Set<T | U>;
```

Comment:
We've made this specific to `Set` rather than a "read-only set" to avoid creating a global until this has landed in typescript.

See https://github.com/denoland/deno/issues/22851#issuecomment-2021674314